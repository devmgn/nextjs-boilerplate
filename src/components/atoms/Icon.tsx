import styled from '@emotion/styled';
import ExampleSvg from '@/assets/example.svg';

type StyledIconProps = {
  size?: number | undefined;
};

export const Icon = styled.i<StyledIconProps>`
  width: ${({ size }) => (size !== undefined ? `${size}px` : '24px')};
  height: ${({ size }) => (size !== undefined ? `${size}px` : '24px')};

  > svg {
    width: inherit;
    height: inherit;
  }
`;

type IconProps = React.ComponentProps<typeof Icon>;

export const ExampleIcon: React.FC<IconProps> = (props) => {
  return (
    <Icon {...props}>
      <ExampleSvg />
    </Icon>
  );
};
