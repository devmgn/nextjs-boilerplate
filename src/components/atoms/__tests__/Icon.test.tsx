import { render } from '@testing-library/react';
import { ExampleIcon } from '../Icon';

describe('Icon', () => {
  test('スナップショットテスト', () => {
    const { asFragment } = render(<ExampleIcon />);
    expect(asFragment()).toMatchSnapshot();
  });
});
