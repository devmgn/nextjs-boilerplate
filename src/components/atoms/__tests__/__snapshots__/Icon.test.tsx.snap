// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon スナップショットテスト 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 24px;
  height: 24px;
}

.emotion-0>svg {
  width: inherit;
  height: inherit;
}

<i
    class="emotion-0 emotion-1"
  >
    <svg />
  </i>
</DocumentFragment>
`;
